app/src/main/java/com/omkarmoghe/pokemap/views/MainActivity.java

```diff
@@ -170,6 +170,12 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
     }
 
     private void startNotificationService(){
+
+        // check for if the service is already running
+        if (PokemonNotificationService.isRunning()) {
+            stopNotificationService();
```

Comment:
True, this is a quick fix, not a refactoring; sadly the reality is that the service is started/stopped onResume/onPause in MainActivity. I just think it's better to put a quick fix now, than let the app spawn as many processes as the user interacts with the app